app/javascript/components/supplemental_application/sections/Lease.js

```diff
@@ -0,0 +1,168 @@
+import React from 'react'
+import { filter, map, isEmpty } from 'lodash'
+
+import Button from '~/components/atoms/Button'
+import ContentSection from '~/components/molecules/ContentSection'
+import FormGrid from '~/components/molecules/FormGrid'
+import InlineModal from '~/components/molecules/InlineModal'
+import formUtils from '~/utils/formUtils'
+import ParkingInformationInputs from './ParkingInformationInputs'
+import RentalAssistance from './RentalAssistance'
+
+import { pluck } from '~/utils/utils'
+import { withContext } from '../context'
+import { CurrencyField, Label, SelectField } from '~/utils/form/final_form/Field.js'
+import { MultiDateField } from '~/utils/form/final_form/MultiDateField'
+import { validateLeaseCurrency } from '~/utils/form/validations'
+
+const toggleNoPreferenceUsed = (form, event) => {
+  // lease.preference_used need to be reset, otherwise SF validation fails
+  if (isEmpty(event.target.value)) {
+    form.change('lease.preference_used', '')
+  }
+  form.change('lease.no_preference_used', isEmpty(event.target.value))
+}
+
+const LeaseActions = ({ onSave, onCancel, onDelete, isNew, loading }) => (
+  <div className='form-grid_item column'>
+    <Button
+      classes='primary margin-right'
+      small
+      onClick={onSave}
+      disabled={loading}
+      text='Save Lease'
+    />
+    <Button
+      classes='secondary'
+      small
+      onClick={onCancel}
+      disabled={loading}
+      text='Cancel'
+    />
+    {!isNew && (
+      <Button
+        classes='alert-fill right'
+        small
+        onClick={onDelete}
+        disabled={loading}
+        text='Delete'
+      />
+    )}
+  </div>
+)
+
+const Lease = ({ form, submitting, values, store }) => {
+  const {
+    availableUnits,
+    application,
+    handleSaveLease,
+    handleDeleteLease,
+    handleClickCancelLease
+  } = store
+  const availableUnitsOptions = formUtils.toOptions(
+    map(availableUnits, pluck('id', 'unit_number'))
+  )
+  const noUnitsOptions = [{ value: '', label: 'No Units Available' }]
+  const confirmedPreferences = filter(application.preferences, {
+    post_lottery_validation: 'Confirmed'
+  })
+  const confirmedPreferenceOptions = formUtils.toOptions(
+    map(
+      [{ id: null, preference_name: 'None' }, ...confirmedPreferences],
+      pluck('id', 'preference_name')
+    )
+  )
+
+  const getVisited = (fieldName) => (
+    form.getFieldState(fieldName)?.visited
```

Comment:
I totally thought `?.` was only a typescript thing, that's great

Reply:
I feel like I try in every environment I'm in and am always thrilled when it exists.